crypto/fipsmodule/digest/digest.c

```diff
@@ -269,8 +281,16 @@ int EVP_Digest(const void *data, size_t count, uint8_t *out_md,
 
   EVP_MD_CTX_init(&ctx);
   ret = EVP_DigestInit_ex(&ctx, type, impl) &&
-        EVP_DigestUpdate(&ctx, data, count) &&
-        EVP_DigestFinal_ex(&ctx, out_md, out_size);
+        EVP_DigestUpdate(&ctx, data, count);
+  if (EVP_MD_flags(type) & EVP_MD_FLAG_XOF) {
+    if (out_size == NULL) {
+      OPENSSL_PUT_ERROR(DIGEST, ERR_R_PASSED_NULL_PARAMETER);
+      return 0;
+    }
+    ret &= EVP_DigestFinalXOF(&ctx, out_md, *out_size);
+  } else {
+    ret &= EVP_DigestFinal_ex(&ctx, out_md, out_size);
```

Comment:
`EVP_DigestFinalXOF` cleans up the context and `EVP_DigestFinal_ex` doesn't but the context is cleaned in line 294. Why not use `EVP_DigestFinal` instead and remove line 294?

Reply:
I was preserving prior code, but I agree that your suggested changes are functionally equivalent and cleaner. will update.